src/Site.php

```diff
@@ -188,17 +188,30 @@ public function wp_object(): ?WP_Site
     /**
      * Switches to the blog requested in the request
      *
-     * @param string|integer|null $site_name_or_id
+     * @param string|integer|null $blog_identifier The name or ID of the blog to switch to. If `null`, the current blog.
      * @return integer with the ID of the new blog
      */
-    protected static function switch_to_blog($site_name_or_id)
+    protected static function switch_to_blog($blog_identifier)
     {
-        if ($site_name_or_id === null) {
-            $site_name_or_id = \get_current_blog_id();
+        $current_id = \get_current_blog_id();
+
+        if ($blog_identifier === null) {
+            $blog_identifier = $current_id;
+        }
+
+        $info = \get_blog_details($blog_identifier);
+
+        if (false === $info) {
+            return $current_id;
         }
-        $info = \get_blog_details($site_name_or_id);
-        \switch_to_blog($info->blog_id);
-        return $info->blog_id;
+
+        $blog_identifier = $info->blog_id;
+
+        if ($current_id !== $blog_identifier) {
+            \switch_to_blog($blog_identifier);
+        }
+
+        return $blog_identifier;
```

Comment:
Make sure we always return an integer.

```suggestion
        return (int) $blog_identifier;
```

Reply:
Since I am quite new to the whole typecasting thing, a small question.

Is it a possibility to typecast `$blog_identifier` at https://github.com/timber/timber/blob/43acfc764d6d96f60ab49fedf36ce784f788e832/src/Site.php#L208

instead of on line 210 and 214 as the return type. If so what would be preferred?

Reply:
Oh yeah, you could use:

```php
$blog_identifier = (int) $info->blog_id;
```

I think that would be better than using the typecasts later.

The basic problem is that when `$blog_identifier` is passed to the function, then it could be a string:

```php
$site->switch_to_blog( '3' );
```

And when we compare a string to a number, `===` will always return `false`. In the future, we could prevent that by requiring an integer to be passed, which could be `null` as well:

```
protected static function switch_to_blog( ?int $blog_identifier = null)
```

But I don’t want to add that breaking change just yet.